### ISSUE TITLE
Unicode equal comparison failed to convert both arguments to Unicode

### ISSUE THREAD

**Issue by dmulter** [PyCQA/pylint#1426]
### Steps to reproduce
1. Create a file named ```bug.py``` with the following lines:
```
#!/usr/bin/env python
# -*- coding: latin-1 -*-
'''Test pylint bug'''

countries = {
    'Åland Islands' # pylint: disable=missing-final-newline
}
```
2. Run ```pylint bug.py```

### Current behavior
- Output is:
```
/Users/.../python2.7/site-packages/pylint/checkers/format.py:1012: UnicodeWarning: Unicode equal comparison failed to convert both arguments to Unicode - interpreting them as being unequal
  if line[-1] in unsplit_ends:
```

### Expected behavior
- There should be no warning message output by pylint and the code should report perfect.
- I also shouldn't have to add the disable for ```missing-final-newline```
- This also appears to be new behavior since I upgraded to pylint 1.7

### pylint --version output
```
pylint 1.7.1,
astroid 1.5.2
Python 2.7.10 (default, Oct 23 2015, 19:19:21)
[GCC 4.2.1 Compatible Apple LLVM 7.0.0 (clang-700.0.59.5)]
```
